test/Mono.Linker.Tests.Cases/LinkAttributes/LinkerAttributeRemovalConditional.cs

```diff
@@ -0,0 +1,91 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Reflection;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+
+namespace Mono.Linker.Tests.Cases.LinkAttributes
+{
+	[SetupLinkAttributesFile ("LinkerAttributeRemovalConditional.xml")]
+	[IgnoreLinkAttributes (false)]
+	class LinkerAttributeRemovalConditional
+	{
+		public static void Main ()
+		{
+			Kept_1 ();
+			Kept_2 ();
+			Kept_3 ();
+			Kept_4 ();
+			Removed_1 ();
+			Removed_2 ();
+		}
+
+		[Kept]
+		[KeptAttributeAttribute (typeof (TestConditionalRemoveAttribute))]
+		[TestConditionalRemove ()]
+		static void Kept_1 ()
+		{
+		}
+
+		[Kept]
+		[KeptAttributeAttribute (typeof (TestConditionalRemoveAttribute))]
+		[TestConditionalRemove ("my-value", "string value")]
+		static void Kept_2 ()
+		{
+		}
+
+		[Kept]
+		[KeptAttributeAttribute (typeof (TestConditionalRemoveAttribute))]
+		[TestConditionalRemove (1, true)]
+		static void Kept_3 ()
+		{
+		}
+
+		[Kept]
+		[KeptAttributeAttribute (typeof (TestConditionalRemoveAttribute))]
+		[TestConditionalRemove ("remove", 1)]
```

Comment:
Please add a comment why this is not getting removed - the only reason I found possible was that the parameter is typed `object` in this case, which probably doesn't match the `string` we have in the XML. If that's the case I find this rather weird, because the values match exactly.

Reply:
Good observation. I changed the tests to be easier to understand.